.github/workflows/ci.yml

```diff
@@ -561,73 +561,10 @@ jobs:
-      - cpu
-      - os-family=Linux
-    env:
-      ANDROID_ABI: "arm64-v8a"
-      BUILD_DIR: ${{ needs.build_all.outputs.build-dir }}
-      BUILD_DIR_ARCHIVE: ${{ needs.build_all.outputs.build-dir-archive }}
-      BUILD_DIR_GCS_ARTIFACT: ${{ needs.build_all.outputs.build-dir-gcs-artifact }}
-    steps:
-      - name: "Checking out repository"
-        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
-      - name: "Checking out runtime submodules"
-        run: ./build_tools/scripts/git/update_runtime_submodules.sh
-      - name: "Downloading build dir archive"
-        run: gcloud alpha storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
-      - name: "Extracting install from build dir archive"
-        run: tar -xf "${BUILD_DIR_ARCHIVE}" "${BUILD_DIR}/install"
-      - name: "Building for Android"
-        run: |
-          build_tools/github_actions/docker_run.sh \
-            --env "ANDROID_ABI=${ANDROID_ABI}" \
-            --env "IREE_HOST_BINARY_ROOT=${BUILD_DIR}/install" \
-            gcr.io/iree-oss/android@sha256:76c2a52dcd6d07601227b965ac87d021c1d2d5e2d01f46ad58da28c89267f2ab \
-            build_tools/cmake/build_android.sh
-
-  baremetal_riscv32:
-    needs: [setup, build_all]
-    if: needs.setup.outputs.should-run == 'true'
-    runs-on:
-      - self-hosted  # must come first
-      - runner-group=${{ needs.setup.outputs.runner-group }}
-      - environment=${{ needs.setup.outputs.runner-env }}
-      - cpu
-      - os-family=Linux
-    env:
-      BUILD_RISCV_DIR: "build-riscv32-baremetal"
-      BUILD_ARCH: "rv32-baremetal"
-      BUILD_DIR: ${{ needs.build_all.outputs.build-dir }}
-      BUILD_DIR_ARCHIVE: ${{ needs.build_all.outputs.build-dir-archive }}
-      BUILD_DIR_GCS_ARTIFACT: ${{ needs.build_all.outputs.build-dir-gcs-artifact }}
-    steps:
-      - name: "Checking out repository"
-        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
-      - name: "Checking out runtime submodules"
-        run: ./build_tools/scripts/git/update_runtime_submodules.sh
-      - name: "Downloading build dir archive"
-        run: gcloud alpha storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
-      - name: "Extracting install from build dir archive"
-        run: tar -xf "${BUILD_DIR_ARCHIVE}" "${BUILD_DIR}/install"
-      - name: "Cross-compiling and testing riscv32 baremetal"
-        run: |
-          ./build_tools/github_actions/docker_run.sh \
-            --env "RISCV_ARCH=${BUILD_ARCH}" \
-            --env "BUILD_RISCV_DIR=${BUILD_RISCV_DIR}" \
-            --env "BUILD_PRESET=test" \
-            --env "IREE_HOST_BINARY_ROOT=${BUILD_DIR}/install" \
-            gcr.io/iree-oss/riscv@sha256:d6f0e293a50faf5abbd564c1d1bb9dc6456d7ce93d07b131c381fa64c1daed62 \
-            bash -euo pipefail -c \
-            "./build_tools/cmake/build_riscv.sh && tests/riscv32/smoke.sh"
 
-  linux_riscv:
+  # emscripten is not in the test matrix because it is set as
```

Comment:
I think we could include emscripten and add reference to the matrix in the if stage. Something like

```
if: needs.setup.outputs.should-run == 'true' && contains(matrix.ci_stages, needs.setup.outputs.ci-stage)
...
- platform: emscripten
  ci_stages: ['postsubmit']
```

Reply:
Interesting. Let me try it out.

Reply:
Strategy matrix is not visible at `if` yet
https://github.com/iree-org/iree/actions/runs/3382670562

We can make a separated step to render this matrix, but we can also just wait for emscripten ready for presubmit CI.

Reply:
Oh don't worry about it then